tests/drivers/i2s/i2s_api/src/test_i2s_errors.c

```diff
@@ -0,0 +1,166 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/kernel.h>
+#include <zephyr/ztest.h>
+#include <zephyr/drivers/i2s.h>
+#include "i2s_api_test.h"
+
+ZTEST_USER(i2s_errors, test_i2s_improper_configuration)
+{
+	int err;
+	struct i2s_config invalid_config = { .word_size = 17U,
+					     .channels = 2U,
+					     .format = I2S_FMT_DATA_FORMAT_I2S,
+					     .frame_clk_freq = FRAME_CLK_FREQ,
+					     .block_size = BLOCK_SIZE,
+					     .timeout = TIMEOUT,
+					     .options = I2S_OPT_FRAME_CLK_MASTER |
+							I2S_OPT_BIT_CLK_MASTER,
+					     .mem_slab = &tx_mem_slab };
+
+	err = i2s_configure(dev_i2s, I2S_DIR_TX, &invalid_config);
+	zassert_not_equal(err, 0, "I2S configuration did not detect improper word size (17)");
```

Comment:
Driver API does not document such value.

Reply:
The excerpt you copied from Zephyr documentation (you could have marked it as quote so it wouldn't get converted to the bullet list) doesn't in any way restrict the allowed word sizes. The only word size related thing that can be derived from it, is that 24 bits words will be padded to 32 bits.

Currently there is no configuration option for how the samples are stored in memory (i.e. if 24 bit samples should be padded to 32-bit or not). Adding this would require stable API change but this is probably to be done by someone who stumbles on this problem due to their actual hardware needs (I see no much point for doing it "just-in-case" at this point).

I think this check is bogus.

Reply:
This case is removed